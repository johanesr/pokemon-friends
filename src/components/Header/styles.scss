hr {
    border: 5px solid black;
    margin-top: 0px;
  }

header {
  position: sticky;
  top: 0;
  background-color: #ff1c1c;

  .logo-pokemon {
    margin-right: auto;
    height: 100px;
    @media (min-width: 992px) {
      width: 200px;
    }
    @media (max-width: 992px) {
      width: 150px;
    }
    padding: 10px 5%;
  }
}

@media (min-width: 992px) {
  ul {
    position: absolute;
    right: 10px;
    top: 15px;

    li {
      list-style: none;
      display: inline-block;
      float: right;
      font-weight: 700;
      padding: 20px 20px;
      margin: 0px 20px;
      background-color: #000000;
      border-radius: 5px;
      box-shadow: 5px 5px 5px #000000;
      transition: 1s;
    }

    li:hover {
      //padding: 30px 30px;
      transform: scale(1.2,1.2);
    }

    a {
      color: #ffffff;
      text-decoration: none;
    }
  }
}

@media (max-width: 991px) {
  .toggle-button {
    display: inline-block;
    position: absolute;
    top: 35px;
    right: 15px;
    background-color: #ffffff;
    padding: 5px;
    border-radius: 5px;
    cursor: pointer;
    box-shadow: 5px 5px 5px #000000;

    [class*=bar1], [class*=bar2], [class*=bar3] {
      align-content: space-between;
      width: 35px;
      height: 5px;
      background-color: #333;
      margin: 6px 0px;
      transition: 0.5s;
    }

    .bar1open {
      transform: rotate(-45deg) translate(-5px, 5px);
    }

    .bar2open {
      display: none;
    }
    .bar3open {
      transform: rotate(45deg) translate(-1px, -5px);
    }
  }

  nav{
    width: 100%;
    position: relative;

    ul {
      position: absolute;
      right: 0;
      //display: none; //can't animate display
      height: 0;
      opacity: 0;
      margin: 0;
      text-align: center;
      transition: 0.3s ease-in-out;

      li {
        width: 100%;
        border-radius: 0;
        list-style: none;
        display: none;
        float: right;
        font-weight: 700;
        padding: 20px 20px;
        background-color: #000000;
      }

      a {
        text-decoration: none;
        color: #ffffff;
      }
      a:active {
        color: #ff1c1c;
      }
    }
    .open {
      //display: block;
      height: 100%;
      opacity: 1;
      li {
        display: block;
      }
    }
  }
}