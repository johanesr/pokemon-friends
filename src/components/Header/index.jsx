import './styles.scss';

function Header() {
  return (
    <div className="header-wrapper">
      <img src="https://upload.wikimedia.org/wikipedia/commons/9/98/International_Pok%C3%A9mon_logo.svg" alt="Pokemon-Logo"/>
      <div>My Favorite Pokemons</div>
      <hr/>
    </div>
  );
}

export default Header;