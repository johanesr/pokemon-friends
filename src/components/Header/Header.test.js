import { shallow } from 'enzyme';
import Header from './index';

const wrapper = shallow(<Header />);

it("renders header correctly", () => {
  expect(wrapper).toMatchSnapshot(); // check if a component renders correctly
});

it('renders burger button click correctly', () => {
  const burgerButton = wrapper.find('.toggle-button')
  burgerButton.simulate('click');
  expect(wrapper).toMatchSnapshot();
});