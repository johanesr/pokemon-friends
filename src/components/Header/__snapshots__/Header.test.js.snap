// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders burger button click correctly 1`] = `
<Fragment>
  <header>
    <Link
      to="/"
    >
      <img
        alt="Pokemon-Logo"
        className="logo-pokemon"
        src="https://upload.wikimedia.org/wikipedia/commons/9/98/International_Pok%C3%A9mon_logo.svg"
      />
    </Link>
    <div
      className="toggle-button"
      onClick={[Function]}
    >
      <div
        className="bar1open"
      />
      <div
        className="bar2open"
      />
      <div
        className="bar3open"
      />
    </div>
    <nav>
      <ul
        className="open"
      >
        <li>
          <Link
            to="/list"
          >
            Pokemon List
          </Link>
        </li>
        <li>
          <Link
            to="/my-list"
          >
            My Pokemons
          </Link>
        </li>
      </ul>
    </nav>
    <hr />
  </header>
</Fragment>
`;

exports[`renders header correctly 1`] = `
<Fragment>
  <header>
    <Link
      to="/"
    >
      <img
        alt="Pokemon-Logo"
        className="logo-pokemon"
        src="https://upload.wikimedia.org/wikipedia/commons/9/98/International_Pok%C3%A9mon_logo.svg"
      />
    </Link>
    <div
      className="toggle-button"
      onClick={[Function]}
    >
      <div
        className="bar1"
      />
      <div
        className="bar2"
      />
      <div
        className="bar3"
      />
    </div>
    <nav>
      <ul
        className=""
      >
        <li>
          <Link
            to="/list"
          >
            Pokemon List
          </Link>
        </li>
        <li>
          <Link
            to="/my-list"
          >
            My Pokemons
          </Link>
        </li>
      </ul>
    </nav>
    <hr />
  </header>
</Fragment>
`;
